lms/djangoapps/bookmarks/views.py

```diff
@@ -0,0 +1,301 @@
+                    "user_message": _(u"Invalid usage id")
+                },
+                status=status.HTTP_400_BAD_REQUEST
+            )
+        return Response(
+            BookmarkSerializer(bookmark, context={"fields": DEFAULT_FIELDS + OPTIONAL_FIELDS}).data,
+            status=status.HTTP_201_CREATED
+        )
+
+
+class BookmarksDetailView(APIView):
+    """
+    **Use Cases**
+
+        Get or Delete a specific bookmark.
+
+    **Example Requests**:
+
+        GET /api/bookmarks/v1/bookmarks/{username},{usage_id}/?fields=path&display_name
+
+        DELETE /api/bookmarks/v1/bookmarks/{username},{usage_id}/
+
+    **Response for GET**
+        Users can only get their own bookmarks
+
+        * id: String. The identifier string for the bookmark: {user_id},{usage_id}.
+
+        * course_id: String. The identifier string of the bookmark's course.
+
+        * usage_id: String. The identifier string of the bookmark's XBlock.
+
+        * display_name: (optional) String. Display name of the XBlock.
+
+        * path: (optional) List of dicts containing {"usage_id": "", display_name:""} for the XBlocks
+            from the top of the course tree till the parent of the bookmarked XBlock.
+
+        * created: ISO 8601 String. The timestamp of bookmark's creation.
+
+    **Response for DELETE**
+        Users can only delete their own bookmarks
+
+        A successful delete returns a 204 and no content.
+
+        Users can only delete their own bookmarks. If the requesting user
+        does not have username "username", this method will return with a
+        status of 404, Thee same is true in case if a bookmark does not exist.
+
+    """
+    authentication_classes = (OAuth2Authentication, SessionAuthentication)
+    permission_classes = (permissions.IsAuthenticated,)
+
+    serializer_class = BookmarkSerializer
+
+    def get(self, request, username=None, usage_id=None):
+        """
+        GET /api/bookmarks/v1/bookmarks/{username},{usage_id}?fields=path&display_name
+        """
+        if request.user.username != username:
+            # Return a 404. If one user is looking up the other users.
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        optional_fields_param = self.request.QUERY_PARAMS.get('fields', [])
+        optional_fields = optional_fields_param.split(',') if optional_fields_param else []
+        optional_fields_to_add = DEFAULT_FIELDS + [field for field in optional_fields if field in OPTIONAL_FIELDS]
+
+        try:
+            usage_key = UsageKey.from_string(usage_id)
+        except InvalidKeyError as exception:
+            log.error(exception.message)
+            return Response(
+                {
+                    "developer_message": exception.message,
+                    "user_message": _(u"Invalid usage id")
+                },
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        try:
+            bookmarks_data = get_bookmark(
```

Comment:
Nit: this is only a single bookmark, create? I would expect the variable name to be "bookmark_data".